q/gradle-quickstart.sh

```diff
@@ -100,53 +103,49 @@ rootProject.name = '${projectName}'
 EOF
 
 cat > build.gradle <<EOF
-buildscript {
-    repositories {
-        jcenter() // this applies only to the Gradle 'Shadow' plugin
-    }
-    dependencies {
-        classpath 'com.github.jengelman.gradle.plugins:shadow:2.0.4'
-    }
-}
-
 plugins {
     id 'java'
     id 'application'
     // shadow plugin to produce fat JARs
-    id 'com.github.johnrengelman.shadow' version '2.0.4'
+    id 'com.github.johnrengelman.shadow' version '7.1.2'
 }
 
-
-// artifact properties
-group = '${organization}'
-version = '${version}'
-mainClassName = '${organization}.StreamingJob'
-description = """Flink Quickstart Job"""
-
 ext {
     javaVersion = '1.8'
     flinkVersion = '${flinkVersion}'
     scalaBinaryVersion = '${scalaBinaryVersion}'
-    slf4jVersion = '1.7.7'
-    log4jVersion = '1.2.17'
+    slf4jVersion = '1.7.32'
+    log4jVersion = '2.17.1'
+    flinkVersionNew = flinkVersion.toString().replace("-SNAPSHOT", "") >= "1.15"
 }
 
+// artifact properties
+group = '${organization}'
+version = '${version}'
+if (flinkVersionNew) {
+    mainClassName = '${organization}.DataStreamJob'
+} else {
+    mainClassName = '${organization}.StreamingJob'
+}
+description = """Flink Quickstart Job"""
 
 sourceCompatibility = javaVersion
 targetCompatibility = javaVersion
 tasks.withType(JavaCompile) {
-	options.encoding = 'UTF-8'
+    options.encoding = 'UTF-8'
 }
 
 applicationDefaultJvmArgs = ["-Dlog4j.configuration=log4j.properties"]
```

Comment:
```suggestion
applicationDefaultJvmArgs = ["-Dlog4j.configuration=log4j2.properties"]
```

This appears to be out of date. [See here](https://github.com/apache/flink/tree/master/flink-quickstart/flink-quickstart-java/src/main/resources/archetype-resources/src/main/resources)

Reply:
Fixed, we needed also `-Dlog4j.configurationFile` instead of `-Dlog4j.configuration` to make it work with log4j2